packages/cli/src/lib/setup/setupVendor.ts

```diff
@@ -0,0 +1,277 @@
+import type { Client as ObjectsRedisClient } from '@iobroker/db-objects-redis';
+import { tools } from '@iobroker/js-controller-common';
+import fs from 'fs-extra';
+import deepClone from 'deep-clone';
+import { isDeepStrictEqual } from 'util';
+
+export interface CLIVendorOptions {
+    objects: ObjectsRedisClient;
+}
+
+const VENDOR_FILE = '/etc/iob-vendor.json';
+
+export class Vendor {
+    private readonly objects: ObjectsRedisClient;
+
+    constructor(options: CLIVendorOptions) {
+        this.objects = options.objects;
+    }
+
+    /**
+     * Deep merge of source into target object
+     *
+     * @param target target obj
+     * @param source source obj
+     */
+    private deepMerge(target: Record<string, any>, source: Record<string, any>): void {
+        Object.keys(source).forEach(attr => {
+            if (typeof source[attr] === 'object' && !(source[attr] instanceof Array)) {
+                if (target[attr] && typeof target[attr] !== 'object') {
+                    target[attr] = {};
+                }
+                target[attr] = target[attr] || {};
+                this.deepMerge(target[attr], source[attr]);
+            } else {
+                target[attr] = source[attr];
+            }
+        });
+    }
```

Comment:
I could swear we have similar code somewhere else already.

Reply:
you maybe mean the `extendCommon` fun in `setupUpload`. This has some more logic, maybe it's possible to get them into a single fun. However I am not sure if this would be desireable, via a short grep there is no exact same function by function name or a single line I have tested.